NetworkingServiceKit/Classes/Networking/AlamoNetworkManager.swift

```diff
@@ -193,29 +191,29 @@ open class AlamoNetworkManager: NetworkManager {
     ///
     /// - Parameter data: data stream
     /// - Returns: a response, empty dictionary if there were issues parsing the data
-    private func errorResponse(fromData data: Data?, request: URLRequest?) -> MSErrorDetails? {
-        var errorResponse: MSErrorDetails? = nil
+    private func errorDetails(fromData data: Data?, request: URLRequest?, code: Int, error: Error?) -> MSErrorDetails {
         let components = requestComponents(request)
         let body = components?.body
         let path = components?.path
         
+        var errorMessage: String?
+        
         if let responseData = data,
             let responseDataString = String(data: responseData, encoding:String.Encoding.utf8),
             let responseDictionary = self.convertToDictionary(text: responseDataString) {
 
             if let responseError = responseDictionary["error"] as? [String: Any],
-                let errorType = responseError["type"] as? String,
                 let message = responseError["message"] as? String {
-                errorResponse = MSErrorDetails(errorType: errorType, message: message, body: body, path: path)
+                errorMessage = message
             } else if let responseError = responseDictionary["errors"] as? [[String: Any]],
                 let responseFirstError = responseError.first,
-                let errorType = responseFirstError["label"] as? String,
                 let message = responseFirstError["message"] as? String {
                 //multiple error
-                errorResponse = MSErrorDetails(errorType: errorType, message: message, body: body, path: path)
+                errorMessage = message
             }
         }
-        return errorResponse
+        
+        return MSErrorDetails(message: errorMessage ?? (error?.localizedDescription ?? ""), body: body, path: path, code: code, underlyingError: error)
```

Comment:
Should we make it more clear to the developer if we couldn't parse an errorMessage? Maybe add a default message?